src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetRuleSubscriberBase.cs

```diff
@@ -79,7 +79,9 @@ public void AddSubscriptions(AggregateCrossTargetProjectContext newProjectContex
         public void ReleaseSubscriptions()
         {
             _currentProjectContext = null;
-            _subscriptions.Dispose();
+
+            _subscriptions?.Dispose();
+            _subscriptions = null;
```

Comment:
As @abpiskunov pointed out, I *think* these are multi-threaded, so both this and the assignment in the creation would need to be locked, or `Interlocked`.

Reply:
No, the type does not support concurrent calls to `AddSubscriptions`/`ReleaseSubscriptions`. If it is _supposed_ to (and I don't think it is) then we need to change a lot more than this and we should handle that as a separate change.